test/common.py

```diff
@@ -105,18 +106,52 @@ def signed_to_int(x: int, xlen: int) -> int:
     return x | -(x & (2 ** (xlen - 1)))
 
 
+def guard_nested_collection(cont: Any, t: Type[T]) -> TypeGuard[_T_nested_collection[T]]:
+    if isinstance(cont, list):
+        return all([guard_nested_collection(elem, t) for elem in cont])
+    elif isinstance(cont, dict):
+        return all([guard_nested_collection(elem, t) for elem in cont.values()])
```

Comment:
Not sure if we want Yet Another Dict Tree Traversal... I kind of think this can also be done in a tiny bit more generic way, like:
```suggestion
    if isinstance(cont, (list, dict)):
        if isinstance(cont, dict):
            cont = cont.values()
        return all([guard_nested_collection(elem, t) for elem in cont])
```

we should just do one generic version of it, and pass a callback/aggregator function along.

Reply:
> we should just do one generic version of it, and pass a callback/aggregator function along.

As for now we don't have dict/list tree traversal in any other place, so i decided to don't make it separate function as for now. In future if this will be needed, it can be done separate.